indexer/validate_and_format_contacts.cpp

```diff
@@ -406,4 +406,30 @@ bool ValidateLinePage(string const & page)
   return (domain == "line.me" || strings::EndsWith(domain, ".line.me"));
 }
 
+bool isSocialContactTag(string_view const tag)
+{
+  return (tag == "contact:instagram" || tag == "contact:facebook" || tag == "contact:twitter"
+          || tag == "contact:vk" || tag == "contact:line");
+}
+
+// Functions ValidateAndFormat_{facebook,instagram,twitter,vk}(...) by default strip domain name
+// from OSM data and user input. This function prepends domain name if it's needed.
+string denormalizeSocialContact(string_view tag, string_view value)
```

Comment:
This function converts `string_view value` to `string` even if no changes are made. Don't know how to improve it.